nx-X11/programs/Xserver/hw/nxagent/Keyboard.c

```diff
@@ -1858,16 +1858,17 @@ void nxagentKeycodeConversionSetup(void)
     char *sessionpath = nxagentGetSessionPath();
     if (sessionpath != NULL)
     {
-      int keyboard_file_path_size = strlen(sessionpath) + strlen("/keyboard");
-      char *keyboard_file_path = malloc((keyboard_file_path_size + 1) * sizeof(char));
+      char *keyboard_file_path = NULL;
       FILE *keyboard_file;
-      if (keyboard_file_path == NULL)
+
+      if ((asprintf(&keyboard_file_path, "%s/keyboard", sessionpath) == -1))
```

Comment:
Nope, we can't use `asprintf` yet. It's not your fault, but really mine since I didn't notice this error when reviewing some of @sunweaver's commits that used `asprintf`.

It doesn't mean that you have to change the code - but this PR will be blocked until I've backported `os/xprintf.c` so that `asprintf` is available across platforms.

Reply:
what platforms are lacking it?

"     These functions are GNU extensions, not in C or POSIX.  They are also
       available under *BSD.  The FreeBSD implementation sets strp to NULL
       on error."

Reply:
By the way: backporting asprintf doesn't sound like fun: http://insanecoding.blogspot.de/2014/06/memory-management-in-c-and-auto.html

Reply:
I don't know what platforms are not providing it exactly, but there certainly seem to be X.Org-supported ones that don't.

Backporting `asprintf` is really easy! But only because X.Org already did it. I'll take a stab at it tomorrow (or at a later time) after I went through the other open PRs.